### ISSUE TITLE
`nix-env -i` always creates new profile link outside of /nix/var/nix/profiles/per-user

### ISSUE THREAD

**Issue by iwinux** [NixOS/nix#2430]
Hi,

I just set up Nix (2.1.1, single-user, channel: https://nixos.org/channels/nixpkgs-unstable) on my macOS 10.12.6, installed a few packages, and then wiped it all out accidentally via `nix-collect-garbage -d` 🤒. 

Seems that all those profile links were created directly in `$HOME`:

```sh
$ ls /nix/var/nix/profiles/per-user/$USER/ -l
total 32
lrwxr-xr-x 1 limbo staff 15 Sep 18 00:24 channels -> channels-1-link
lrwxr-xr-x 1 limbo staff 60 Sep 17 22:45 channels-1-link -> /nix/store/2wkxpc5jza977bks1rsj5irm29r0ljsv-user-environment
```

I reinstalled Nix again but still no profile links created in `/nix/var/nix/profiles/per-user/$USER/`, and I'm pretty sure `nix-collect-garbage -d` will wipe it out again....

I've found [this issue](https://github.com/NixOS/nix/issues/2070) but it was fixed a while ago? What might goes wrong?

**Comment by LnL7** [comment]
What does ~/.nix-profile point to, did you run nix upgrade-nix at some point before 2.1?

**Comment by iwinux** [comment]
To answer your questions:

1. it pointed to `$HOME/.nix-profile-link-N` where N is 1, 2, 3, 4...
2. this is a fresh install so no `upgrade-nix` involved

-----------

But....after some debugging I found the culprit in my own zsh configs: I exported an env named `NIX_PROFILE`, which obviously has some special meaning to Nix and confused Nix to put profile links outside of `/nix/var/nix/profiles` 😓.

After renaming it to `MY_NIX_PROFILE` everything works fine now.
